platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java

```diff
@@ -377,4 +399,18 @@ public static <T> Consumer<T> combineConsumers(final List<Consumer<T>> consumers
             }
         };
     }
+
+    /**
+     * Returns a string representation of the given byte count in human readable format.
+     *
+     * @param bytes number of bytes
+     * @return human-readable string representation of the given byte count
+     */
+    public static String byteCountToDisplaySize(final long bytes) {
+        final SimplifiedQuantity<DataUnit> simplifiedQuantity = DataUnit.UNIT_BYTES.simplify(bytes);
+        final double quantity = simplifiedQuantity.quantity();
+        final Unit<?> unit = simplifiedQuantity.unit();
+
+        return unit.buildFormatter(quantity).setDecimalPlaces(1).render();
+    }
```

Comment:
A unit formatter automatically chooses a good unit for the provided quantity, so no need to do that manually.

```suggestion
    public static String byteCountToDisplaySize(final long bytes) {
        return UNIT_BYTES.buildFormatter(bytes)
            .setDecimalPlaces(1)
            .render();
    }
```